cardano-config/src/Cardano/Config/Types.hs

```diff
@@ -123,14 +125,26 @@ data CardanoConfiguration = CardanoConfiguration
     , ccWallet              :: !Wallet
     } deriving (Eq, Show)
 
-data NodeCLI = NodeCLI
+data NodeRealCLI = NodeRealCLI
+    { realMscFp :: !MiscellaneousFilepaths
+    , realNodeAddr :: !NodeAddress
+    , realConfigFp :: !ConfigYamlFilePath
+    , realTraceOpts :: !TraceOptions
+    , validateDB :: !Bool
+    }
```

Comment:
When are we planning to introduce differences between the options for the real and mock modes? Many of the options are specific to either real or mock.

For example the `(ncProtocol nc)` when we run the node is an example where for the real mode the choice is pre-determined, but for the mock more there is a choice of mock/test protocols. So the mock side would have a field for protocol choice that the real side does not have, since there is only one choice for the real side.

Reply:
So the first differentiation would be removing `NodeId` from the `configuration.yaml` file and moving that into a command line argument for mock protocols.

Re: `(ncProtocol nc)`: If I understand what you are saying,  `(ncProtocol nc)` can give any of the protocols: https://github.com/input-output-hk/cardano-node/blob/52e6fb4befd5c89e7a3923166358f0f5a829c573/cardano-config/src/Cardano/Config/Types.hs#L407

Some need to be re-named however to match e.g `BFT`: https://github.com/input-output-hk/cardano-node/blob/04894d0082921ced7f478631add4191af3374af5/cardano-config/src/Cardano/Config/Protocol.hs#L120

I'm not sure if I've answered your concern.